### ISSUE TITLE
Linter mistakenly thinks a random function is a JSX class.

### ISSUE THREAD

**Issue by dapetcu21** [yannickcr/eslint-plugin-react#316]
<img width="1030" alt="screen shot 2015-11-18 at 10 55 17" src="https://cloud.githubusercontent.com/assets/428060/11236528/06af9566-8de3-11e5-93f7-40f328bd5b5b.png">

**Comment by epmatsw** [comment]
The reason it's getting flagged is that the `display-name` rule treats functions that return JSX as stateless react components ([link](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/display-name.md#about-component-detection)), since they sort of technically could be. You might be able to get around it by turning on the `acceptTranspilerName` option. This is yet another case where it might be good to have an `ignoreStatelessFunctionComponents` option, but I'm not sure what @yannickcr's thoughts on that are.

Tangential question: what font is that? I love the === and => rendering.

**Comment by dapetcu21** [comment]
It's tonsky/FiraCode :)

**Comment by dapetcu21** [comment]
About stateless components:
I see the value in having an ignore option, though there are certain rules that simply make no sense with stateless components (like `display-name`)

**Comment by yannickcr** [comment]
`display-name` is still useful if you want to prevent "anonymous" components like:
```js
export default function(props) {
  return <div>Hello {props.name}</div>
}
```

> This is yet another case where it might be good to have an ignoreStatelessFunctionComponents option, but I'm not sure what @yannickcr's thoughts on that are.

I do not really see the advantage over just turning on `acceptTranspilerName`.

Also, in the next major version the `acceptTranspilerName` option will be `on` by default.

**Comment by dapetcu21** [comment]
Still, that's not the only rule where you might want to have `ignoreStatelessFunctionComponents`.

In the example above, the intent is to write a react-native `Navigator.NavigationBar` route mapper. `eslint-plugin-react` thinks those are stateless components even though they aren't.

Since it's a bit hard to differentiate stateless components from just functions returning JSX, what would be the preferred way of disabling other checks like `react/no-multi-comp` in these situations?
